src/runtype.ts

```diff
@@ -85,6 +85,10 @@ export interface Runtype<A = unknown> {
    * Adds a brand to the type.
    */
   withBrand<B extends string>(brand: B): Brand<B, this>;
+  withBrand<B extends string, T extends Static<this>>(
+    brand: B,
+    constraint: ConstraintCheck<this>,
+  ): Brand<B, Runtype<T>>;
```

Comment:
Seems like the overload should just be

```ts
  withBrand<B extends string>(
    brand: B,
    constraint: ConstraintCheck<this>,
  ): Brand<B, this>;
```

What's the rationale for this overload though? Is it really that much of a time saver to be able to write

```ts
Number.withBrand('someNumber').withConstraint(x => x > 3)
```

instead of

```ts
Number.withBrand('someNumber', x => x > 3)
```

? And more importantly, what makes brands special that they should have a shortcut for constraints that other runtypes don't? The nice thing about the `withConstraint` combinator is that it's simple and universal; you don't have to wonder if this or that runtype has such a shortcut. Sometimes conceptual simplicity is worth sacrificing a small amount of concision.

Reply:
I think a brand type without constraint checks should not exist: we create a unique type with (I presume) special semantic but don't prove it by a code - looks wrong for me. So when we spread out brand and constraint definitions it pushes us to mistake.

Reply:
>I think a brand type without constraint checks should not exist

A frequent use case for brands that I'm aware of is units of measure or currencies... there's no runtime constraint you can write to validate that a plain `number` represents for example meters instead of degrees fahrenheit.

Reply:
Okay, that's a controversial question, let's leave it